orchestra/static/orchestra/common/components/checklist/js/directives.js

```diff
@@ -0,0 +1,243 @@
+        // completed and the new order is set.
+        drake.on('drag', function() {
+          scope.preventAction = true;
+          scope.$apply();
+        });
+        drake.on('dragend', function() {
+          setOrder();
+          scope.preventAction = false;
+          scope.$apply();
+        });
+
+        // Determines and saves the order changed by drag event
+        var setOrder = function() {
+          var items = el.get(0).querySelectorAll('.checklist-item');
+
+          for (var i = 0; i < items.length; i++) {
+            var item = items[i];
+            getItemByKey(item.getAttribute('data-key')).order = i;
+          }
+        }
+
+        // Retrieves checklist item based on angular object key
+        var getItemByKey = function(key) {
+          var match;
+          var item;
+          for (var i = scope.data.items.length - 1; i >= 0; i--) {
+            item = scope.data.items[i];
+            if (item.$$hashKey === key) {
+              return item;
+            }
+          };
+        }
+
+        // Add new checklist item
+        scope.addItem = function() {
+          scope.data.items.push({'title': '',
+                                 'comments': [{'author': scope.author, 'text': ''}],
+                                 'checked': false,
+                                 'expanded': false,
+                                 'readonly': true,
+                                 'order': scope.data.items.length});
+        }
+
+        // Remove checklist item
+        scope.removeItem = function(item, confirm_needed) {
+          var index = scope.data.items.indexOf(item);
+          if (!confirm_needed || confirm('Are you sure you want to delete this item?')) {
+            scope.preventAction = true;
+            scope.data.items.splice(index, 1);
+            scope.data.deletedItems.push(item);
+
+            // Reset order after $digest block (once item has been removed)
+            $timeout(function() {
+              setOrder();
+              scope.preventAction = false;
+            }, 0, false);
+          }
+        }
+
+        // Allow checklist title to be edited
+        scope.editItem = function(item, $event) {
+          if (item.readonly) {
+            // Item will become editable
+            if ($event) {
+              // Set focus on input
+              $event.target.focus();
+            }
+            scope.data.items.forEach(function(scopeItem) {scopeItem.readonly = true})
+          }
+          item.readonly = !item.readonly;
+        }
+
+        // Expand checklist item to show comments
+        scope.expandItem = function(item) {
+          item.expanded = !item.expanded;
+          scope.data.items.forEach(function(scopeItem) {scopeItem.readonly = true})
+        }
+
+        // Return localized comment timestamp
+        scope.getCommentTimestamp = function(comment) {
+          return new Date(comment.timestamp).toLocaleString();
```

Comment:
Handled by the browser, should be local time: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString

There are definitely more sophisticated ways of handling this, but we'd need to pick a third-party library—let me know if this is something you're imminently worried about.